umbraco/Umbraco-CMS
# Bugfix: Fixed issues when changing UmbracoPath in config

This PR fixes issues that was shown when changing UmbracoPath in config. + Enabled possibility to copy the wwwroot/umbraco into another folder.


## Test on this branch:
- Change umbraco path and and other relevant configs:
 ```
 {
  "Umbraco": {
    "CMS": {
      "Global": {
        "ReservedPaths": "~/app_plugins/,~/install/,~/mini-profiler-resources/,~/my-secret-loginpanel/,",
        "UmbracoPath": "~/my-secret-loginpanel",
        "IconsPath": "~/my-secret-loginpanel/assets/icons",
      }
    }
  }
}
 ```
 - Rename the `wwwroot/umbraco` folder to `wwwroot/my-secret-loginpanel`
- Run umbraco.

## Test on publish output
- `run the build.ps1 script`
- install the template from the build.out folder (`dotnet new -i Umbraco.Templates::9.0.0-* --nuget-source={PathToProject}\build.out`)
- install using template: `dotnet new umbraco -n TestUmbracoPathChange`
- Add this line into a propertyGroup element of the csproj file
`<UmbracoWwwrootName>my-secret-loginpanel</UmbracoWwwrootName>`
- Change the configs
 ```
 {
  "Umbraco": {
    "CMS": {
      "Global": {
        "ReservedPaths": "~/app_plugins/,~/install/,~/mini-profiler-resources/,~/my-secret-loginpanel/,",
        "UmbracoPath": "~/my-secret-loginpanel",
        "IconsPath": "~/my-secret-loginpanel/assets/icons",
      }
    }
  }
}
 ```
 - Build 
    - Verify the output is genered in wwwroot/my-secret-loginpanel
  - Run umbraco and verify backoffice exists in `/my-secret-loginpanel` instead of `/umbraco`
